queries.sql

```diff
@@ -8,3 +8,30 @@ SELECT name,escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species is NULL;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT delete_date;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO delete_date;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT MAX(escape_attempts) FROM animals WHERE neutered = true OR neutered = false;
+SELECT MIN(weight_kg),MAX(weight_kg) FROM animals;
```

Comment:
- [ ] Kindly fix these two select statements to match the requirement:
   - `Who escapes the most, neutered or not neutered animals?`.
   - `What is the minimum and maximum weight of each type of animal?`.

### Hint: 
```
use the 'GROUP BY' clause
```

Reply:
Thanks for pointing out to improve my code , It was helpful to learn.Thanks for your time and efforts